core/controllers/collection_viewer.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2015 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Controllers for the Oppia collection learner view."""
+
+__author__ = 'Ben Henning'
+
+from core.controllers import base
+from core.domain import collection_domain
+from core.domain import collection_services
+from core.domain import config_domain
+from core.domain import exp_domain
+from core.domain import exp_services
+from core.domain import rights_manager
+from core.platform import models
+(user_models,) = models.Registry.import_models([models.NAMES.user])
+import feconf
+import utils
+
+def require_collection_playable(handler):
+    """Decorator that checks if the user can play the given collection."""
+    def test_can_play(self, collection_id, **kwargs):
+        """Check if the current user can play the collection."""
+        if rights_manager.Actor(self.user_id).can_play(
+                rights_manager.ACTIVITY_TYPE_COLLECTION, collection_id):
+            return handler(self, collection_id, **kwargs)
+        else:
+            raise self.PageNotFoundException
+
+    return test_can_play
+
+
+class CollectionPage(base.BaseHandler):
+    """Page describing a single collection."""
+
+    PAGE_NAME_FOR_CSRF = 'collection'
+
+    @require_collection_playable
+    def get(self, collection_id):
+        """Handles GET requests."""
+        try:
+            collection = collection_services.get_collection_by_id(
+                collection_id)
+        except Exception as e:
+            raise self.PageNotFoundException(e)
+
+        version = collection.version
+
+        if not rights_manager.Actor(self.user_id).can_view(
```

Comment:
Technically not, since this is checking can_view versus require_collection_playable checking can_play. That being said, I'm a bit confused as to the difference. I'm not sure which one is correct here.